TSOClient/tso.client/UI/Profile/UIJobInfo.cs

```diff
@@ -0,0 +1,377 @@
+            Level.CaptionStyle.Size = 12;
+            Level.CaptionStyle.Shadow = false;
+            Level.Alignment = TextAlignment.Right | TextAlignment.Middle;
+            this.Add(Level);
+            verticalSpace += 12 + standardVerticalSpace + standardVerticalSpace;
+
+            if (!jobInformation.MaxLevel)
+            {
+                PromotionRequirements = new UILabel
+                {
+                    Position = new Vector2(30, verticalSpace),
+                    Size = new Vector2(200, 16),
+                    CaptionStyle = TextStyle.DefaultTitle
+                };
+                PromotionRequirements.CaptionStyle = PromotionRequirements.CaptionStyle.Clone();
+                PromotionRequirements.CaptionStyle.Color = Color.White;
+                PromotionRequirements.CaptionStyle.Size = 16;
+                PromotionRequirements.CaptionStyle.Shadow = true;
+                PromotionRequirements.Alignment = TextAlignment.Left | TextAlignment.Middle;
+                PromotionRequirements.Caption = "Promotion Requirements:";
+                this.Add(PromotionRequirements);
+                verticalSpace += 16 + standardVerticalSpace;
+
+                UIImage PerformanceSectionImage = new UIImage(GetTexture((ulong)0x7A400000001)).With9Slice(13, 13, 13, 13);
+                PerformanceSectionImage.SetSize(360, sectionHeight);
+                PerformanceSectionImage.Position = new Vector2(20, verticalSpace);
+                Add(PerformanceSectionImage);
+                this.Add(PerformanceSectionImage);
+
+                int performanceTitleSectionHeight = (sectionHeight / 2) - 8;
+
+                PerformanceTitle = new UILabel
+                {
+                    Position = new Vector2(40, verticalSpace + performanceTitleSectionHeight),
+                    Size = new Vector2(200, 12),
+                    CaptionStyle = TextStyle.DefaultTitle
+                };
+                PerformanceTitle.CaptionStyle = PerformanceTitle.CaptionStyle.Clone();
+                PerformanceTitle.CaptionStyle.Color = new Color(238, 247, 169);
+                PerformanceTitle.CaptionStyle.Size = 14;
+                PerformanceTitle.CaptionStyle.Shadow = false;
+                PerformanceTitle.Alignment = TextAlignment.Left | TextAlignment.Middle;
+                PerformanceTitle.Caption = "Performance";
+                this.Add(PerformanceTitle);
+
+                int progressBarWidth = 145;
+                ProgressBar = new UIProgressBar()
+                {
+                    X = 360 - progressBarWidth,
+                    Y = verticalSpace + performanceTitleSectionHeight - 5,
+                    Value = 0
+                };
+                ProgressBar.SetSize(progressBarWidth, 27);
+                ProgressBar.Caption = "";
+                
+                ProgressBar.Background = TextureGenerator.GenerateRoundedRectangle(GameFacade.GraphicsDevice, new Color(37, 58, 77), progressBarWidth, 27, 9);
```

Comment:
This is a bespoke texture, so perhaps you should store a private reference to it that you can `.Dispose()` during the `OnRemoved()` override below.

Reply:
Calling `.Dispose()` on `ProgressBar.Background.Dispose()` during the Remove event. I hope this accomplishes the same goal.